addons/mail/static/tests/chatter/web/chatter_topbar.test.js

```diff
@@ -138,38 +139,40 @@ test("attachment counter with attachments", async () => {
 });
 
 test("attachment counter while loading attachments", async () => {
+    const def = new Deferred();
     const pyEnv = await startServer();
     const partnerId = pyEnv["res.partner"].create({});
-    onRpc("/mail/data", async (request) => {
-        const { params } = await request.json();
-        if (params.fetch_params.some((fetchParam) => fetchParam[0] === "mail.thread")) {
-            await new Deferred(); // simulate long loading
-        }
+    const { waitStoreFetch } = listenStoreFetch("mail.thread", async () => {
+        asyncStep("before mail.thread");
+        await def;
     });
     await start();
     await openFormView("res.partner", partnerId);
     await contains("button[aria-label='Attach files']");
     await advanceTime(DELAY_FOR_SPINNER);
     await contains("button[aria-label='Attach files'] .fa-spin");
     await contains("button[aria-label='Attach files']", { count: 0, text: "0" });
+    await waitForSteps(["before mail.thread"]);
+    def.resolve();
+    await waitStoreFetch("mail.thread");
 });
 
 test("attachment counter transition when attachments become loaded", async () => {
+    const def = new Deferred();
     const pyEnv = await startServer();
     const partnerId = pyEnv["res.partner"].create({});
-    const deferred = new Deferred();
-    onRpc("/mail/data", async (request) => {
-        const { params } = await request.json();
-        if (params.fetch_params.some((fetchParam) => fetchParam[0] === "mail.thread")) {
-            await deferred;
-        }
+    const { waitStoreFetch } = listenStoreFetch("mail.thread", async () => {
+        asyncStep("before mail.thread");
```

Comment:
Note: this step is necessary to ensure the callback is actually applied, it makes the test much more robust.